IPython/html/nbextensions.py

```diff
@@ -242,6 +242,62 @@ def install_nbextension(path, overwrite=False, symlink=False, user=False, prefix
             src = path
             _maybe_copy(src, full_dest, verbose)
 
+
+def make_cmdclass(setupfile, path, enable=None):
+    """Build nbextension cmdclass dict for the setuptools.setup method.
+
+    Parameters
+    ----------
+    setupfile: str
+        Path to the setup file.
+    path: str
+        Directory relative to the setup file that the nbextension code lives in.
+    enable: [str=None]
+        Extension to "enable".  Enabling an extension causes it to be loaded
+        automatically by the IPython notebook.
+
+    Usage
+    -----
+    setup(
+        name='flightwidgets',
+        ...
+        cmdclass=make_cmdclass(__file__, 'flightwidgets', 'flightwidgets/init'),
+    )
+    """
+
+    from setuptools.command.install import install
+    from setuptools.command.develop import develop
+    from os.path import dirname, abspath, join
+
+    from .services.config import ConfigManager
+
+    def run_nbextension_install(develop):
+        install_nbextension(join(dirname(abspath(setupfile)), path), symlink=develop)
+        if enable is not None:
+            print("Enabling the extension ...")
+            cm = ConfigManager()
+            cm.update('notebook', {"load_extensions": {enable: True}})
```

Comment:
I think I'd be more conservative with kernel extensions, because they affect what code you can run - if someone uses a non-standard magic function or something in a notebook, I want to be able to see where it's coming from. Also, it's easy to enable kernel extensions with a command inside that kernel.

Server and frontend extensions are, to my mind, more like extensions to a GUI program than modules for a programming language. I think there should be a technical separation between 'installed' and 'enabled', to allow things like systemwide installed extensions that users can choose to enable, but from the perspective of a user wanting an extension, the easiest action should be 'install and enable'.